server/routes/completeOrArchive/completeOrArchiveGoalController.ts

```diff
@@ -0,0 +1,41 @@
+import type { RequestHandler } from 'express'
+import createError from 'http-errors'
+import type { CompleteOrArchiveGoalForm } from 'forms'
+import CompleteOrArchiveGoalView from './completeOrArchiveGoalView'
+import EducationAndWorkPlanService from '../../services/educationAndWorkPlanService'
+
+export default class CompleteOrArchiveGoalController {
+  constructor(private readonly educationAndWorkPlanService: EducationAndWorkPlanService) {}
+
+  getCompleteOrArchiveGoalView: RequestHandler = async (req, res, next): Promise<void> => {
+    const { prisonNumber, goalReference } = req.params
+    const { prisonerSummary } = res.locals
+
+    const actionPlan = await this.educationAndWorkPlanService.getActionPlan(prisonNumber, req.user.username)
```

Comment:
What you've done here is fine, and follows the patterns/approaches in other similar controllers 👍 

I've been thinking for a while that this pattern of "get the entire action plan, then find the goal in question from the returned array" is a bit meh 🤷‍♂️ and we can perhaps do better.

From memory I think it was because originally we never had an API endpoint to get a goal by its reference - all we could do was get the entire action plan, hence the UI code that we have.
But I _think_ since then we might have implemented a suitable API endpoint to get a single goal by it's reference, so we might be able to use that 🤔 

One for another day, but perhaps we should identify which pages/controllers this might affect, then create some tickets around it

Reply:
yes it is a bit overkill grabbing everything when all we want is the title